app/models/miq_filter.rb

```diff
@@ -174,8 +174,12 @@ def self.apply_belongsto_filters(inputs, bfilters)
           end
           break if filtered.last == p
         else
-          vcmeta = belongsto2object(tag)
+          vcmeta_list = belongsto2object_list(tag)
+          vcmeta_list.pop if vcmeta_list.last.kind_of?(Host)
+          vcmeta = vcmeta_list.last
```

Comment:
@jrafanie after my fix https://github.com/lpichler/manageiq/commit/222da54d991abe84a06066b496bbf20e8120c031 method `belongsto2object_list` is now returning `Host` object also unlike as it was before, so it is transforming path:
`/belongsto/ExtManagementSystem|X1/EmsFolder|X2/EmsFolder|X2/EmsFolder|host/Host|HOST1` to objects:
`[EmsFolder(name=X1), EmsFolder(name=X2), EmsFolder(name=host), Host(name=HOST1) ]`
and before it was just:
`[EmsFolder(name=X1), EmsFolder(name=X2), EmsFolder(name=host)]`

`belongsto2object` takes last object from resullt of the `belongsto2object_list`.

`apply_belongsto_filters(inputs, filters)`  `inputs` are objects(Host for example) and `filters` are belongsto filters and it is returning objects from inputs which are matching `filters`

before when it was called: 
`apply_belongsto_filters([Host(name="H2")], [...EmsFolder|X2/EmsFolder|host/Host|H2])`
it was returning `[Host(name="H2")]` thanks to ancestor(`vcmeta.is_ancestor_of?(p)` condition here https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_filter.rb#L179
because `vcmeta = belongsto2object(tag)` was object EmsFolder (host folder, representing penultimate patern of filter).
After this fix(https://github.com/lpichler/manageiq/commit/222da54d991abe84a06066b496bbf20e8120c031) this line was returning object `Host(name=H2)' and ancestor condition was not able to match it. 

`apply_belongs_to_filters` is called in `authorized_for_user?` https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/filterable_mixin.rb#L50 and when `apply_belongs_to_filters` was returning `[]` then  `authorized_for_user?` was returning `false` here https://github.com/ManageIQ/manageiq/blob/master/app/controllers/application_controller/tree_support.rb#L265 and it did not allow to display whole tree of hosts in group's filters in tab Host and Cluster in this part as it was before.

So this change here basically reflects this fix https://github.com/lpichler/manageiq/commit/222da54d991abe84a06066b496bbf20e8120c031 and it is ensuring previous behavior.